lib/versioning/rez/index.ts

```diff
@@ -0,0 +1,324 @@
+const matchVersion = new RegExp(
+  `^(?<version>${versionGroup})$`
+); /* Match a version number (e.g. 1.0.0) */
+const exactVersion = new RegExp(
+  `^(?<exact_version>==(?<exact_version_group>${versionGroup})?)$`
+); /* Match an exact version number (e.g. ==1.0.0) */
+// inclusiveBound is called inclusive but behaviour in rez is this:
+// package-1..3 will match versions 1.2.3, 2.3.4, but not 3.0.0 or above
+const inclusiveBound = new RegExp(
+  `^(?<inclusive_bound>(?<inclusive_lower_version>${versionGroup})?\\.\\.(?<inclusive_upper_version>${versionGroup})?)$`
+); /* Match an inclusive bound (e.g. 1.0.0..2.0.0) */
+// Add ? after |\\+) in order to match >=1.15
+const lowerBound = new RegExp(
+  `^(?<lower_bound>(?<lower_bound_prefix>>|>=)?(?<lower_version>${versionGroup})?(\\k<lower_bound_prefix>|\\+)?)$`
+); /* Match a lower bound (e.g. 1.0.0+) */
+const upperBound = new RegExp(
+  `^(?<upper_bound>(?<upper_bound_prefix><(?=${versionGroup})|<=)?(?<upper_version>${versionGroup})?)$`
+); /* Match an upper bound (e.g. <=1.0.0) */
+// Add ,? to match >=7,<9 (otherwise it just matches >=7<9)
+const ascendingRange = new RegExp(
+  `^(?<range_asc>(?<range_lower_asc>(?<range_lower_asc_prefix>>|>=)?(?<range_lower_asc_version>${versionGroup})?(\\k<range_lower_asc_prefix>|\\+)?),?(?<range_upper_asc>(\\k<range_lower_asc_version>,?|)(?<range_upper_asc_prefix><(?=${versionGroup})|<=)(?<range_upper_asc_version>${versionGroup})?))$`
+); /* Match a range in ascending order (e.g. 1.0.0+<2.0.0) */
+// Add , to match <9,>=7 (otherwise it just matches <9>=7)
+const descendingRange = new RegExp(
+  `^(?<range_desc>(?<range_upper_desc>(?<range_upper_desc_prefix><|<=)?(?<range_upper_desc_version>${versionGroup})?(\\k<range_upper_desc_prefix>|\\+)?),(?<range_lower_desc>(\\k<range_upper_desc_version>,|)(?<range_lower_desc_prefix><(?=${versionGroup})|>=?)(?<range_lower_desc_version>${versionGroup})?))$`
+); /* Match a range in descending order (e.g. <=2.0.0,1.0.0+) */
+
+function getVersionParts(input: string): [string, string] {
+  const versionParts = input.split('-');
+  if (versionParts.length === 1) {
+    return [input, ''];
+  }
+
+  return [versionParts[0], '-' + versionParts[1]];
+}
+
+function padZeroes(input: string): string {
+  if (/[~^*]/.test(input)) {
+    // ignore ranges
+    return input;
+  }
+
+  const [output, stability] = getVersionParts(input);
+
+  const sections = output.split('.');
+  while (sections.length < 3) {
+    sections.push('0');
+  }
+  return sections.join('.') + stability;
+}
+
+function plus2npm(input: string): string {
+  if (input.includes('+')) {
+    return '>=' + input.replace('+', ' ');
+  }
+  return input;
+}
+
+function rez2npm(input: string): string {
+  if (matchVersion.test(input)) {
+    return input;
+  }
+  if (exactVersion.test(input)) {
+    return input.replace('==', '=');
+  }
+  if (inclusiveBound.test(input)) {
+    return '>=' + input.replace('..', ' <');
```

Comment:
Is it expected to only replace the first occurence of `..`

Reply:
rez doesn't support multiple dot ranges. You can argue replacing only the first occurrence is good enough. Let me know how you want to proceed.

Reply:
leave as is